yjit/src/asm/x86_64/mod.rs

```diff
@@ -970,6 +970,25 @@ impl CodeBlock
             _ => unreachable!()
         };
     }
+
+    /// Encode a conditional move instruction
+    fn write_cmov(&mut self, opcode1: u8, dst: X86Opnd, src: X86Opnd) {
+        if let X86Opnd::Reg(reg) = dst {
+            match src {
+                X86Opnd::Reg(_) => (),
+                X86Opnd::Mem(_) => (),
+                _ => unreachable!()
+            };
+
+            assert!(reg.num_bits >= 16);
+            let sz_pref = reg.num_bits == 16;
+            let rex_w = reg.num_bits == 64;
+
+            self.write_rm(sz_pref, rex_w, dst, src, 0xff, 2, &[0x0f, opcode1]);
+        } else {
+            unreachable!()
+        }
+    }
```

Comment:
Looks good but reiterating my comment that these helper methods should not live on the CodeBlock